### ISSUE TITLE
[Tracking] Custom Report Names on New Expensify

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#44340]
# [DESIGN DOC ➡️](https://docs.google.com/document/d/1U2AiMOlWvT144a2WVi2XCSc18piyE_iFrv7cYWhsAyw/edit?usp=sharing)

## Proposal
**Custom report names on New Expensify**

**Strategy:** New Expensify aims to capture the untapped small business expense management market through an intuitive product that spreads virally from employees to company decision makers, and between companies and friends via word of mouth consensus as the best tool for the job. To achieve this goal it must quickly solve customer problems and simplify their life, providing a seamless experience to become an indispensable tool that is sharp and reliable.

**Background:** Classic and New Expensify allow workspace admins to configure "Custom Report Names" using a formula made up from any text and various "formula parts" such as `{report:type}`, `{report:startdate}`, etc. Some parts allow selecting various subfields like `{report:submit:from:firstname}`
 and functions can be applied to all parts, such as `{report:submit:from:email|frontPart}`. For a full list of supported formula parts, see the report level formulas and the functions on this [help page](https://help.expensify.com/articles/expensify-classic/spending-insights/Export-Expenses-And-Reports#formulas).

On Classic, when the formula is updated to a new value and a user loads the reports page, all non-chat reports loaded by the app that have not been approved yet, and do not have a manually set name, will have their name recomputed according to the formula and saved to the database. There is also a setting to prevent users from manually setting a report name. Formulas can reference data at the report level, transaction, policy, and user levels. When referenced data is updated and a report is loaded again on classic, the name will be updated with the new value (in most cases). Users have very likely come to expect this behavior since it has worked this way for many years.

Report titles are one of our most popular features. Over 82% of admins change the default to something more specific. Custom report names are useful at all levels of approval and for admins, as it helps give context to what kinds of expenses are being grouped together.

New Expensify is built on a foundation of realtime chat, and all information it displays updates without requiring any manual page refreshes. Users can go offline and use the app, confident that everything will save and any missed updates will load when they go back online. Reports are not loaded from the database when refreshing the page manually or navigating to the inbox page. Instead, reports are loaded from the server once on sign in and thereafter updates to the reports and other data are sent from the backend as soon as they occur. If the app misses updates for any reason, it will fetch them from the server. Due to that architecture, when data is changed on the server it will queue all updates that are required to put the users' UI into the correct state.

Classic requires refreshing the page to update almost all data and when doing so it is almost always completely reloaded from the database.

**Problem:** When a user saves a custom report name formula in their workspace settings on New Expensify, reports in New Expensify will not have their names set based on the formula, which causes confusion, prevents the user from trusting our pre-accounting system, and prevents them from having their reports named in a uniform manner with important information.

**Solution:** Make custom report names work on New Expensify by doing the following.
In progress or already done:

- Move report name computation from Web-E to Auth so that names can be computed following our 1:1:1 rule
- Computing the name in Web-E would add 3+ Auth requests to every API command that could affect the name formula, which is many commands.
- Move a few formula parts at a time to Auth, deploying each part to production but hiding it behind a very limited "beta"
- Start with the most commonly used formula parts, based on querying usage for users billed in the last 3 months
- Update the computed report name from Auth whenever the report updates, then similarly for transactions
- [Implement custom report field formula parts](https://github.com/Expensify/Expensify/issues/478327)
- Implement the rest of the supported formula parts
- Create a system to [Monitor and resolve differences in auto report title computations ](https://github.com/Expensify/Expensify/issues/478343)between the Web-E and Auth implementations

To do:

- Update custom report names using Bedrock jobs when the policy and other referenced data changes and requires updating many existing reports
- Determine which data should update existing report names, if not all
- Display notes in the UI when custom report names may be stale
- [Plan and implement a system for optimistic report title formula computation](https://github.com/Expensify/App/issues/59689)
- Once there are no undesired differences between the Web-E and Auth implementations, slowly roll it out to more customers, monitoring for any problems
- After it's working well for many customers including large ones, remove the beta

## Tasks

- [x] Post **Proposal (full Problem/Solution statement)** in `#whatsnext`
  - [x] [Posted](https://expensify.slack.com/archives/CC7NECV4L/p1746733065308919) on 5/8/25
- [x] Wait at least one full business day, and until the post has a majority (2/3) of positive reactions (👍)
- [x] Paste Proposal in the space above with a link to the Slack thread
- [x] Email `strategy@expensify.com` and paste in the Proposal
- [x] Host a **pre-design** meeting ([example](https://expensify.slack.com/archives/C01GTK53T8Q/p1665794669539419)) in an appropriate slack channel to discuss any necessary details in public before filling out the **High-level of proposed solution** section.
  - [x] [Posted in #retain](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1747084486181949) on 5/12/15
- [x] Fill out the **High-level overview of the problem**, **Timeline**, **Terminology**, and **High-level of proposed solution** sections of the Design Doc
- [x] Email `strategy@expensify.com` (continue the same email chain as before - your last message should be the WN Proposal) with the link to your Design Doc containing your high-level problem and solution
- [x] Add the `DesignDocReview` label to get the **High-level overview of the problem** and **High-level of proposed solution** section reviewed
- [x] Respond to any questions or concerns and bring up blockers in Slack to get a consensus if necessary
- [x] Confirm that the doc has the minimum necessary number of reviews before proceeding
- [x] Host another pre-design meeting in the appropriate slack channel to ask for engineering feedback on the technical solution.
  - [x] N/A, a pre-design was not deemed necessary
- [x] Fill out the **Detailed implementation of the solution** and related sections.
- [x] Re-add the `DesignDocReview` label to this issue
- [x] Respond to any questions or concerns and bring up blockers in Slack to get consensus if necessary
- [x] Confirm that the doc has the minimum necessary number of reviews before proceeding
- [x] Email `strategy@expensify.com` one last time to let them know the Design Doc is moving into the implementation phase
- [x] Implement the changes
- [x] Add regression tests so that QA can test your feature with every deploy ([instructions](https://stackoverflowteams.com/c/expensify/questions/363))
- [ ] Send out a follow up email to `strategy@expensify.com` once everything has been implemented and do a **Project Wrap-Up** retrospective that provides:
    - Summary of what we accomplished with this project
    - What went well?
    - What could we have done better?
    - What did we learn?

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @neil-marcellini </details>

**Comment by trjExpensify** [comment]
@caitlinwhite1 @thienlnam is this on your radar? Will it need to be `Internal`? 

Adding it to #wave-control in the meantime. CC: @JmillsExpensify

**Comment by thienlnam** [comment]
Is there a photo of what the report title looks like in NewDot?

IIRC this is partially expected. NewDot generates the report title based on the formula optimistically since it could be offline, but the thing it can't support is long report IDs, or anything that requires server side knowledge. 

However, if you are online it should just take the report name. So depending on the circumstances this is likely external

**Comment by heyjennahay** [comment]
If I look at the reports in my online web NewDot they have the incorrect format

![image](https://github.com/Expensify/App/assets/12088484/764166b1-7515-433a-9287-a981cb7dc143)

**Comment by thienlnam** [comment]
Oh gotcha, so it's not just a visual bug in NewDot. 

Hmm, I was testing this on dev and it seems to work alright for me. 
<img width="1268" alt="Screenshot 2024-06-26 at 10 24 12 AM" src="https://github.com/Expensify/App/assets/30609178/36e94a94-0940-4fe1-8b52-7ae7a5fd42a1">

My only other guess is NewDot Test - Expensify a collect policy not a control policy?

**Comment by heyjennahay** [comment]
@thienlnam can you please check your personal NewDot Test reports in NewDot?  To confirm your theory about the Collect v Control workspace being the cause?  My NL Test workspace is Collect but the US Test workspace has been Control for about a month now so if your theory is correct then I would expect to see a difference.

That said, why would the formula not be applied to both Collect and Control workspaces (this feature applies to both types in OldDot today)?  Also if the intention is to require you to upgrade to Control to use Report Formulas then I would still expect the default report title for Collect workspaces to show the short report ID and not the long.  So I still think there is an issue that needs to be resolved.

**Comment by thienlnam** [comment]
I agree there's a bug here, just trying to isolate where the problem might be. It does seem to be a control policy so that rules out that theory.

<img width="966" alt="Screenshot 2024-06-26 at 1 20 03 PM" src="https://github.com/Expensify/App/assets/30609178/45284e9c-2f79-4d88-bc39-4b6bd6a91b73">

So it looks like the report fields are not being returned into this policy - going to have some queries run to see why that is the case. This would be internal
<img width="576" alt="Screenshot 2024-06-26 at 1 24 43 PM" src="https://github.com/Expensify/App/assets/30609178/1193677f-0342-446e-b5b4-a9c408acae00">

**Comment by thienlnam** [comment]
Created an internal GH for a query request [here](https://github.com/Expensify/Expensify/issues/408357)

**Comment by trjExpensify** [comment]
Nice, what are the next steps?

**Comment by trjExpensify** [comment]
Bump on this, @thienlnam.

**Comment by thienlnam** [comment]
Just got back from OOO - got the results of the query back and everything looks as expected.

However the report field is still not getting returned in NewDot so there is likely something wrong in the flow.

Next is investigating the query to see if there's something that might be impacting the return of this data

**Comment by thienlnam** [comment]
Haven't looked into this further yet

**Comment by trjExpensify** [comment]
Any luck?

**Comment by thienlnam** [comment]
Sorry, haven't looked into this further due to instantscan. I've also got app deploy duty this week, so I am hoping to get around to this sometime next week. Feel free to re-assign if needed

**Comment by heyjennahay** [comment]
Noticed this is also happening on reports created on our OldDot production workspace when they are created via Expensify Travel.  Example report ID 6471996360309383

![image](https://github.com/user-attachments/assets/fcf62085-586a-422e-a3ba-ab0a3cecb54d)

Although I just found one which isn't travel or NewDot testing report ID 8929959707960611
![image](https://github.com/user-attachments/assets/aae08e36-d766-4dab-95ba-8d947b023473)

**Comment by trjExpensify** [comment]
@thienlnam putting this customer bug report on your radar as potentially related: https://github.com/Expensify/Expensify/issues/413468

**Comment by thienlnam** [comment]
Looks related, I think the flow is broken on OldDot too

**Comment by trjExpensify** [comment]
Okay, well this might be a CRITICAL bug where we're messing with the report title formulas set on OldDot. CC: @JmillsExpensify

**Comment by thienlnam** [comment]
Okay as I was looking into this again, this has just started to work again. Like all my reports are now created with the correct report name. Are you seeing the same? After Jul 16th all of my reports have started to get the correct reportName

![Screenshot 2024-07-23 at 4 56 43 PM](https://github.com/user-attachments/assets/7ec08309-22ad-4531-b2ba-d02c5ca49738)

**Comment by thienlnam** [comment]
This seems to be resolved, but wanted to get some confirmation

**Comment by trjExpensify** [comment]
@heyjennahay can maybe confirm from accounting perspective on the prod policy. 


Interestingly, I was just testing something and I'm seeing this in the report history: 

<img width="1511" alt="image" src="https://github.com/user-attachments/assets/d9b89d2c-cd95-4611-a57c-28024b09e178">

I didn't take that action myself. Not sure where it's coming from.. 🤔

**Comment by thienlnam** [comment]
Oh yeah looks like we should not generate a policy history when a report is updated from a template value

**Comment by trjExpensify** [comment]
That's a reportAction though on the report, not a workspace change log in #admins.

**Comment by thienlnam** [comment]
Ah whoops, yeah I mean no edited reportAction when a title is updated from a template value on a report

**Comment by trjExpensify** [comment]
Cool, let's address that then. 👍

**Comment by thienlnam** [comment]
There was a conversation about this here - https://expensify.slack.com/archives/C06ML6X0W9L/p1722534299058619 and here https://expensify.slack.com/archives/C06ML6X0W9L/p1722534322032729

Turns out this will require moving the report title template generation to auth

**Comment by thienlnam** [comment]
That sounds good to me

**Comment by thienlnam** [comment]
Updating priority based on [this](https://github.com/Expensify/App/issues/44340#issuecomment-2284546322), we will re-prioritize Auth Report Fields in Nov

**Comment by thienlnam** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
I got assigned this [bug report ](https://github.com/Expensify/App/issues/50018), where now all new reports in NewDot seem to use the long report ID format, rather than including the amount or merchant. Seems kind of related to this issue, but curious if you guys agree?

**Comment by trjExpensify** [comment]
Yeah, that's a dupe I'd say. @thienlnam, where are you on [this](https://expensify.slack.com/archives/C06ML6X0W9L/p1722538645056569?thread_ts=1722534299.058619&cid=C06ML6X0W9L)?

**Comment by thienlnam** [comment]
From [this](https://expensify.slack.com/archives/C06ML6X0W9L/p1722534322032729) convo it seemed de-scoped and deprioritized so I haven't worked on it since. Not sure when we want to re-prioritize it though

**Comment by trjExpensify** [comment]
Hm, I guess I'm not following @JmillsExpensify's comment:

>Keeping them in scope, moving them to auth in a follow-up. Basically just like we did for report next steps.

What exactly is in scope now if this remains broken and blocked until moving them to auth?

**Comment by JmillsExpensify** [comment]
From the linked Slack convo, I was voting for option A, which is this:

> A. Fix the whole report formula inconsistency thing right now i.e. if a formula is updated on either NewDot or OldDot it should update all the reports everywhere. New policies created via NewDot can then set a formula similar to what we are doing in the getMoneyRequestReportName() (but we will have to accept that it won’t be translated). Anything that would cause the name to be recomputed (e.g. amount, report state, etc) should also queue Onyx updates with report name changes. However, the report formula logic stuff is all in PHP at the moment so we would have to queue those updates from there.

cc @marcaaron

**Comment by JmillsExpensify** [comment]
Sounds like [we agreed in Slack](https://expensify.slack.com/archives/C036QM0SLJK/p1728590921571019) that this should be prioritized, so I'm going to put the `Daily` label back on this issue.

**Comment by trjExpensify** [comment]
Should we move it to #quality as suggested in thread?

**Comment by thienlnam** [comment]
@marcaaron has very kindly offered to look into this while I clear some other things off my plate

Marc is going to look into porting this into auth
In the meantime, I will look into the quick fix that doesn't involve that

**Comment by marcaaron** [comment]
Unassigning for now to clear the Overdue off my K2. I can start looking into the report fields stuff when this is more of a priority.

**Comment by thienlnam** [comment]
PR in review to address the quick fix that doesn't involve moving everything to auth at the moment

**Comment by trjExpensify** [comment]
PR is in review, and @iwiznia was back today to take a first pass.

**Comment by trjExpensify** [comment]
@thienlnam how you getting on with that Web-PR?

**Comment by thienlnam** [comment]
I have to move some things to auth and then Web-PR is on hold for that, hopefully will get around to it today or tommorow

**Comment by thienlnam** [comment]
Getting into the final review stages for the Web PR

**Comment by thienlnam** [comment]
Merged, pending deploy

**Comment by trjExpensify** [comment]
Nice! The [Web PR](https://github.com/Expensify/Web-Expensify/pull/44091#issuecomment-2476967310) went to prod on Thurs. Are we done here?

**Comment by heyjennahay** [comment]
To clarify, all new reports created in NewDot since last Thursday should be using the default report title of the workspace?  As it appears we are still using the NewDot default and not the workspace formula

![image](https://github.com/user-attachments/assets/be90a516-3c43-4978-ab61-dc959f950276)
![image](https://github.com/user-attachments/assets/4f62887e-186b-486b-9019-58075a6dbd90)

**Comment by thienlnam** [comment]
This came up [here](https://expensify.slack.com/archives/C07HPDRELLD/p1732594992526249) as well, will re-open to look into it

**Comment by thienlnam** [comment]
Found the issue, it only happens when the report title is not forced. If you force it, the report title gets changed.

Need to update it so that it makes it the new default

**Comment by thienlnam** [comment]
Fix is merged, pending deploy

**Comment by trjExpensify** [comment]
@thienlnam opening an expense report I see this report action which is pretty weird: 

<img width="1508" alt="image" src="https://github.com/user-attachments/assets/97db74b8-f1b0-47f6-a3cd-56c2a6f321ee">

**Comment by francoisl** [comment]
This was reported as a blocker here https://github.com/Expensify/App/issues/53820.

I've got a revert PR on hold here: https://github.com/Expensify/Web-Expensify/pull/44733, though ideally we can find a fix that doesn't reintroduce the previous issue. 
The [original fix PR](https://github.com/Expensify/Web-Expensify/pull/44662) doesn't have any explanation of what it does and why, @thienlnam can you think of an alternative fix? `updateFieldsToMatchPolicy()` hasn't been changed since like 2016, so it's surprising that report titles would have broken just now in NewDot.

**Comment by thienlnam** [comment]
This is likely just a side effect of not having it moved in Auth - basically what we're doing here is we are creating the report with a different default in Auth, and then in web updating the report title based on the default. Let me see if we can skip the message for that easily but I don't know if this needs to be a blocker.

**Comment by thienlnam** [comment]
I suppose what we could try to do here is check to see if the updated report NVP is the report_title, and see if it was previously a default and the policy it is on has a non-default report title? But I'm not entirely sure how the system messages get calculated

**Comment by francoisl** [comment]
The default report title gets computed in Auth [here](https://github.com/Expensify/Auth/blob/29f493df4423f88490a907dff49ac755e0744a83/auth/lib/Report.cpp#L11642-L11658), I was starting to think we'd need to update that method to use the policy's name formula 🤔

**Comment by thienlnam** [comment]
Yeah exactly, we would need to port over the policy formula logic to auth - we have just been avoiding having to do that hence this workaround but it seems to have other issues

**Comment by trjExpensify** [comment]
>IMO it seems a little confusing but not a blocker

I dunno, I think it's pretty bad as it records an action the user never took. That's bad from an audit perspective, as you lose trust in the audit history.

**Comment by JmillsExpensify** [comment]
I tend to agree. I don't think we should be adding history items for actions a user never took. By the way, what's stopping us from moving the formula logic to auth?

**Comment by JmillsExpensify** [comment]
In fact, I had this come up on a report today. Notice how it's saying I renamed the report when I rejected a couple of expensees. 
![CleanShot 2024-12-11 at 17 25 17@2x](https://github.com/user-attachments/assets/b15c9dfe-789b-4c68-a2ca-2885ccf801ab)

**Comment by thienlnam** [comment]
Update:
The short-term change we added didn't work because it started adding policy log messages for each time the report was updated and that was an undesirable update. So now, we are just going to move it to auth. It's going to take a while, and given the holidays I probably won't get back to this until mid Jan if someone else would like to take it over

**Comment by thienlnam** [comment]
^ Same

**Comment by neil-marcellini** [comment]
Taking this over for Jack as [noted here](https://expensify.slack.com/archives/C07NMDKEFMH/p1737133349234189?thread_ts=1737081797.864309&cid=C07NMDKEFMH)

**Comment by trjExpensify** [comment]
Incredible, super helpful.

**Comment by neil-marcellini** [comment]
Here's how the flow works now for my own understanding and future reference.

When you create a report on OldDot it calls CreateReport and passes and empty reportName. The reportName is [generated in Web-E here](https://github.com/Expensify/Web-Expensify/blob/b5990ef81cc49b62c4b9b306a605a76543802efa/lib/ReportStore.php#L371-L381) and set on the report in the Auth command CreateReport.

When you create a money request in NewDot it calls RequestMoney and does not pass a report name. Nor is one passed to the CreateMoneyRequest Auth command. Eventually it calls createExpenseReport and that function [calls createWithAction](https://github.com/Expensify/Auth/blob/00e8a97cb9c2b2c09fd398a911af99f6b25abb9e/auth/lib/Report.cpp#L7712) with an empty report name. That results in a [call to generateDefaultReportName](https://github.com/Expensify/Auth/blob/00e8a97cb9c2b2c09fd398a911af99f6b25abb9e/auth/lib/Report.cpp#L6571-L6575) which is where the current [default format with the reportID comes from](https://github.com/Expensify/Auth/blob/00e8a97cb9c2b2c09fd398a911af99f6b25abb9e/auth/lib/Report.cpp#L11833-L11835).

So we need to implement the OldDot logic from Web-E in Auth Report::generateDefaultReportName, as Francois mentioned earlier.

**Comment by neil-marcellini** [comment]
I got a draft PR started. My approach for now is to write the bare minimum amount of code in Auth to achieve the desired result. There is a large and complex class hierarchy in PHP with a lot of unnecessary code, so I think this makes sense. I'll ask what others think in engineering chat tomorrow.

**Comment by neil-marcellini** [comment]
I'm chugging along on the Auth PR, implementing each specific report title part [as found here](https://help.expensify.com/articles/expensify-classic/spending-insights/Export-Expenses-And-Reports). I worked on this yesterday morning and got [approval from Ionatan](https://expensify.slack.com/archives/C03TQ48KC/p1737643751021349) that my approach without class hierarchies is fine.

This morning I made a good chunk of progress and I plan to work on it more later today.

**Comment by neil-marcellini** [comment]
Yeah I'm checking there too. I assumed they were all implemented in PHP so that's a good call out, thanks.

**Comment by neil-marcellini** [comment]
Ah scratch that about moving ForexUtils::format, luckily [Flo already did that](https://expensify.slack.com/archives/C03TQ48KC/p1713970320124649) as I found from searching Slack. It's called `Currency::format`.

**Comment by neil-marcellini** [comment]
I got a few more formula parts implemented today. I started working on the field formula part which is one of the most complicated, and it's going along well so far.

**Comment by neil-marcellini** [comment]
I made a lot of progress today on the field formula part and I got it tested and working for a basic case when the field value hasn't been set. I'll need to keep testing it a bit more to ensure the more complex cases work.

**Comment by neil-marcellini** [comment]
Yesterday I also made some good progress and got the field formula part test passing. Then I started working on `report:created` and I realized that I need to actually implement Web-E's `getFormattedDate` function that it uses for all dates in formula parts. That required converting from DateJS to SQLite date format.

At this point the PR is getting a bit out of hand, it's a ton of code and there will need to be lots more before the feature is complete. Therefore, I will start breaking the PR up into smaller ones so that it's easier to review and we can make more steady progress.

**Comment by neil-marcellini** [comment]
Ionatan reviewed the first PR and I updated it this morning based on his feedback. Hopefully we'll merge that soon and I'll keep going on the next parts.

**Comment by neil-marcellini** [comment]
After talking with Ionatan, Tom, and Jason we've decided it will be important to come up with a plan for how to break this up in terms of keeping the code focused and also supporting the most common customer use cases.

1. Identify a minimal bug fix for NewDot report creation using the existing Web-E report auto title code
  i. Ask for an exception to the 1:1 rule
2. Query to find the most common auto report title formula parts
3. Implement those formula parts for NewDot report creation in Auth
4. Update the displayed auto title on NewDot when the report updates
5. When the transactions on the report update
6. When the policy updates
7. Implement the rest of the formula parts if we decide it's worthwhile

**Comment by iwiznia** [comment]
What would happen for users that use some of the unimplemented formula parts?

**Comment by neil-marcellini** [comment]
Yesterday Ionatan asked about why the auto title doesn't already work, given that the function `updateFieldsToMatchPolicy` can be called when a report is loaded in the store in Web-E. `updateFieldsToMatchPolicy` is the function that computes the auto title formula and sets the report name. I investigated thoroughly and found that there is already some [code that Jack wrote 3 months ago](https://github.com/Expensify/Web-Expensify/blob/d69450691f2848b1bebf51368cc91300c0eee932/lib/TransactionAPI.php#L1414-L1420) which loads the report for this purpose.

There's a couple problems with the current code. First, the report is already loaded by other function calls before this point so it's already in the cache, and it never calls `updateFieldsToMatchPolicy`. Second, that function doesn't compute the title formula because [this condition](https://github.com/Expensify/Web-Expensify/blob/ce024e8a25096808b949e0d88e70e5305f2ebc2d/lib/Report.php#L2000-L2006) is false, mostly because `$newReport` is defaulted to false. That flag is only set when [creating reports from OldDot](https://github.com/Expensify/Web-Expensify/blob/b5990ef81cc49b62c4b9b306a605a76543802efa/lib/ReportStore.php#L397-L399) via the CreateReport API.

So one option is to call `updateFieldsToMatchPolicy` directly from the requestMoney passing the `$newReport` flag as true. That will compute the report formula which sets the title field nvp from the policy onto the report (write 1), renames the report (write 2), and then updates the title field rNVP with the new computed value (write 3).

Additionally, that [creates a RENAMED report action in Auth](https://github.com/Expensify/Auth/blob/8490ca3104637697d3e5f79717908e99e0712623/auth/command/UpdateReport.cpp#L166-L171) because the "oldName" of the report is not set to the defaultValue of the title field. That could be fixed by always setting it [when creating an expense report in Auth](https://github.com/Expensify/Auth/blob/5c939b6f2ee779d103a93d8dfa4076890a5d380e/auth/lib/Report.cpp#L7701).

However, the real sticking point is that we have many API commands used by NewDot which can create reports. Some of them are RequestMoney, CreateDistanceRequest, TrackExpense, ShareTrackedExpense, SplitBill, etc. I don't want to add 3 extra writes to all of those and we want to move the logic into Auth eventually, so I'll keep going with that.

**Comment by neil-marcellini** [comment]
[[Auth] Compute auto title for expense reports v1](https://github.com/Expensify/Auth/pull/14017) is ready for another round of review now.

**Comment by neil-marcellini** [comment]
I got the v1 Auth PR merged which is very relieving. Next I'm going to work on a smaller chunk and get another PR up for review soon; probably date formatting for formula fields.

**Comment by neil-marcellini** [comment]
I created that issue for identifying the most used formula parts, and added a task list to the issue description that I'll keep up to date. I'm still chipping away at the date formatting. This morning I fixed the date format conversion function and got it working for `{report:created}`, as verified with some basic tests.

**Comment by neil-marcellini** [comment]
This morning I added a bunch of tests for the date format conversion function, more report:created cases such as an invalid date, and I added date formatting to the report:startdate formula part. There's a few other formula parts to add it to and cases to test, and then I should be able to get the date formatting PR up for review.

**Comment by neil-marcellini** [comment]
[Report formula part date formatting](https://github.com/Expensify/Auth/pull/14222#top) that's up for review now

**Comment by neil-marcellini** [comment]
We identified the most used formula parts and I update the to do list in the description to have a more detailed and up to date break down of what's coming next. I'll start on implementing the top 10 most commonly used formula parts.

**Comment by neil-marcellini** [comment]
I have a draft PR up for the top 10 parts. There's actually only 3 that I haven't already implemented which are report:submit, report:policyname, and report:autoreporting. This afternoon I implemented and tested report:policyname, and I implemented report:submit, but I still need to test it.

**Comment by neil-marcellini** [comment]
I just finished testing that report:submitted is working properly for pretty much every case. It had many subfields to test so was extra tricky. Now I'll start implementing `report:autoreporting`

**Comment by neil-marcellini** [comment]
I got `report:autoreporting` fully implemented and now I'm working on testing it. I'll start with tests for AutoReporter::getDateRange like we have in Web-E, because that's the most complex part of it.

**Comment by neil-marcellini** [comment]
This is up for review now! [[HOLD query timing] Implement remaining top 10 formula parts](https://github.com/Expensify/Auth/pull/14230#top)

On to the next part which is implementing the functions that can be added to all formula parts.

**Comment by neil-marcellini** [comment]
The top10 formula part PR is approved and should be deployed soon. I just put the functions PR up for review too. Next up is the `hasCircularReferences` checks.

**Comment by neil-marcellini** [comment]
Actually, I'm going to work on this first "Update the displayed auto title on NewDot when the report updates". The circular reference checks only relate to `field` formula parts and I'm planning to implement those after ensuring the title updates properly when data changes.

 I'll use this formula to start testing `Report to {report:submit:to:email}`, and verify that the title is updated when a report is submitted via NewDot. In Web-E the formulas are updated whenever a report is loaded from the store, but we don't really have that same concept in Auth so I will have to just call a function like Report::updateFormulas whenever a dependency of a formula part changes. Do you have any thoughts on that @iwiznia?

**Comment by neil-marcellini** [comment]
I ran into a complication when trying to test report submitting in Auth using a NewDot flow. When NewDot expense reports are created they are always [set as instant submitted](https://github.com/Expensify/Auth/blob/a3cd38847464e93172a5f475e7ae97a611daafc2/auth/lib/Report.cpp#L7768), but if instant submit is not enabled then they [have an open state](https://github.com/Expensify/Auth/blob/a3cd38847464e93172a5f475e7ae97a611daafc2/auth/lib/Report.cpp#L7752-L7754). In the NewDot UI, we [show the submit button](https://github.com/Expensify/App/blob/e835fbc9c678f31ae39b5bfc58de168c7e3d2e44/src/components/MoneyReportHeader.tsx#L188) if the [report is in an open state](https://github.com/Expensify/App/blob/c7a0ff68a3422820cff4ab6f4937504931ecd517/src/libs/actions/IOU.ts#L8197-L8205), and some other conditions are true which include violation checks.

So in order for the report submit formulas to work properly with NewDot we need to change the logic for when a report is considered submitted in the report formula computation. Or it might be better to align the NewDot submit logic with how it works in OldDot. I'll have to ask for some advice on that.

For now I will test report updates using the formula `Report reimbursementID: {report:reimbursementid}`.

**Comment by trjExpensify** [comment]
Ah, maybe [this proposal](https://expensify.slack.com/archives/CC7NECV4L/p1736182095667549) is going to help you here with the proposed update to instant not "creating reports in the submitted state" but "create open, submit a millisecond later if it's good".

**Comment by neil-marcellini** [comment]
This morning I finished the code and test for verifying that a report will have its name/title updated when it is using the formula part `report:reimbursementid` and it is reimbursed. Now I'm working on testing that manually. Once that's working I'll see if there's any other parts that relate to report updates. If not I'll move on to transaction updates.

I'm also going to add a task to the description to fix determining if a NewDot report is submitted.

**Comment by neil-marcellini** [comment]
I'm working on creating some sub-issues and recruiting others to help this move faster. Short day for me today so I'll keep going on that tomorrow.

**Comment by neil-marcellini** [comment]
Not overdue. After some review feedback I'm making changes to the PR for auto title updates on report changes. Today I'll create some sub-issues for this project.

**Comment by neil-marcellini** [comment]
I added the hot pick label to all unassigned sub-issues.

**Comment by mjasikowski** [comment]
@neil-marcellini FYI I'm planning to port the entire UDF formula/part functionality (with all strings attached) to UDF, so you may want to assign most UDF/Auth related issues to me.

**Comment by neil-marcellini** [comment]
> [@neil-marcellini](https://github.com/neil-marcellini) FYI I'm planning to port the entire UDF formula/part functionality (with all strings attached) to UDF, so you may want to assign most UDF/Auth related issues to me.

Why? I don't think that's valuable. Let's discuss with @iwiznia and decide what the best course of action. What we've currently decided is to leave the report formula stuff as is, not using the report formula class hierarchy, but have it use the UDF object for getting the default value.

**Comment by neil-marcellini** [comment]
I'm working on updating the title for transaction updates now

**Comment by neil-marcellini** [comment]
Not overdue. This morning I finished manual testing. A bunch of the tests are failing probably because I made some changes to how creating accounts works. I'll fix that and then I need to report a bug for NewDot not updating when a transaction is unreported on classic.

**Comment by neil-marcellini** [comment]
The transactions update issue should be closed today when the merged Auth PR is deployed.

Next I will work on [Update NewDot report auto titles when the policy updates](https://github.com/Expensify/Expensify/issues/478317#top). I should probably sync with @mjasikowski about how the custom report fields are coming along since that's something which will need to be updated when the policy updates.

I also create some frontend issues to implement optimistic updates for report formulas.

**Comment by neil-marcellini** [comment]
I was OOO on Friday. I'm going to be reviewing a PR for custom report field formula parts today

**Comment by neil-marcellini** [comment]
I created a couple more issues today, and I'm focusing on this one myself now [Update NewDot report auto titles when the policy updates](https://github.com/Expensify/Expensify/issues/478317)

**Comment by neil-marcellini** [comment]
We're having [a discussion](https://expensify.slack.com/archives/C03TQ48KC/p1744057405769909) about whether auto report titles should update when the policy formula updates, and how that would work.

**Comment by neil-marcellini** [comment]
I'm chatting with Tim and Carlos in a group DM trying to brainstorm what the problem statement should be that will lead us to the right solution to get this completed.

**Comment by neil-marcellini** [comment]
See https://github.com/Expensify/Expensify/issues/478317#issuecomment-2813569656

**Comment by neil-marcellini** [comment]
I've been focused on other priorities, but I hope to get back to this today or tomorrow.

**Comment by neil-marcellini** [comment]
Not overdue, it's Monday morning.

**Comment by neil-marcellini** [comment]
I didn't work on this yesterday because I triaged several issues and investigated a customer issue.

**Comment by neil-marcellini** [comment]
Still working on the plan for updating when the policy changes. Today I'll query to find what the worst case is for updating reports.

**Comment by neil-marcellini** [comment]
The issue body has been updated with out project process template and I posted the proposal [in Slack](https://expensify.slack.com/archives/CC7NECV4L/p1746733065308919).

**Comment by neil-marcellini** [comment]
I got started on the high level doc. I want to add more detail to the high level solution summary before moving on.

**Comment by neil-marcellini** [comment]
I got the solution summary filled out and it feels really solid. I also filled out most of the data storage section, emailed the original proposal to strategy@expensify.com.

Next up I will post a pre-design asking to make sure everyone is aligned about some of the UX decisions I've made in the high level. I hope to send the high level out on Tuesday.

**Comment by neil-marcellini** [comment]
Not overdue, it's Monday morning. Planning to post a pre-design later today.

**Comment by neil-marcellini** [comment]
I posted the [high level predesign](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1747084486181949) in Slack just now.

**Comment by neil-marcellini** [comment]
I put the doc up for high level review yesterday.

**Comment by pecanoro** [comment]
I have read and reviewed this Design Doc!

**Comment by jeremyhollmexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by marcaaron** [comment]
I have read and reviewed this Design Doc!

**Comment by jliexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by ryanschaffer** [comment]
I have read and reviewed this Design Doc!

**Comment by LLPeckham** [comment]
I have read and reviewed this Design Doc!

**Comment by abekkala** [comment]
I have read and reviewed this Design Doc!

**Comment by neil-marcellini** [comment]
Yeah @dukenv0307 this project will fix that

**Comment by cead22** [comment]
I have read and reviewed this Design Doc!

**Comment by nkuoch** [comment]
I have read and reviewed this Design Doc!

**Comment by neil-marcellini** [comment]
I got a pretty sizable chunk of the detailed design written today, about half of it. Tomorrow I will fill out the detailed background, write the query for `getReportIDsForNameComputation`, the remaining phases, etc.

**Comment by anmurali** [comment]
I have read and reviewed this Design Doc!

**Comment by neil-marcellini** [comment]
I finished writing everything for the detailed design. The only part still outstanding is the frontend optimistic design from @sosek108. I'll give him a day or two for that. If absolutely needed I could fill it in myself, but I don't think that will be needed.

**Comment by clarajones-expensify** [comment]
I have read and reviewed this Design Doc!

**Comment by cead22** [comment]
I have read and reviewed this Design Doc!

**Comment by LLPeckham** [comment]
I have read and reviewed this Design Doc!

**Comment by luacmartins** [comment]
I have read and reviewed this Design Doc!

**Comment by neil-marcellini** [comment]
Not overdue, I have been responding to doc comments as they come in.

**Comment by robertjchen** [comment]
I have read and reviewed this Design Doc!

**Comment by neil-marcellini** [comment]
I worked on manual distance stuff a lot so that we can get that design doc out for review. So I don't have time for this today, unfortunately.

**Comment by nkuoch** [comment]
I have read and reviewed this Design Doc!

**Comment by pecanoro** [comment]
I have read and reviewed this Design Doc!

**Comment by neil-marcellini** [comment]
Responding to doc comments now

**Comment by neil-marcellini** [comment]
I got all the comments resolved and I bumped @zsgreenwald Since we need one more review from anyone. I'm sure that will be done in the next few days and we can start implementation whenever

**Comment by neil-marcellini** [comment]
I'm going to start on implementation because I don't think I need to be blocked.

**Comment by neil-marcellini** [comment]
I created all of the issues and asked for some volunteers [here](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1749583993122099?thread_ts=1749570249.246759&cid=C07NZ8B1VTQ)

**Comment by neil-marcellini** [comment]
@mjasikowski picked up some issues and got a bunch of PRs up. I reviewed all 4 of those today. I also made some [progress](https://github.com/Expensify/Expensify/issues/478317#issuecomment-2968381152) on updating all names when the policy changes.

**Comment by neil-marcellini** [comment]
I'm partially OOO. 25% today and 50% the next two days until 6/19 when I will be back fully. I will try to move things along as much as I can in order of priority.

**Comment by neil-marcellini** [comment]
This PR is up for review [[Auth] Return reportIDs when policy report name formula or fields change](https://github.com/Expensify/Auth/pull/15668) and Michal's should be going out soon too

**Comment by neil-marcellini** [comment]
The [Auth pull request](https://github.com/Expensify/Auth/pull/15668) to return the reports that need to be updated has been deployed to production. Yesterday I started working on [implementing the final formula part](https://github.com/Expensify/Auth/pull/15847) that has not been moved to Auth yet. Rachel and I are working on [updating the help site](https://github.com/Expensify/App/pull/64289). I have been working with Michal on displaying the errors for deprecated formula parts, and that [[Web-Expensify] [Hold] Show errors for deprecated formula fields](https://github.com/Expensify/Web-Expensify/pull/47167) is ready to go, but we are waiting to send it out until the messages to admins are ready. We first need to query for which admins to message, and that [query is ready](https://github.com/Expensify/Expensify/issues/513282) to be run by Rafe.

Things are moving along well. Check out the [project board](https://github.com/orgs/Expensify/projects/203/views/1) for a full overview. My top focus on the project right now is implementing the last formula part, ASRReporting. Once that's done, I can get back to working on the bedrock job to update report names.

**Comment by neil-marcellini** [comment]
I completed the implementation of the ComputeReportNames auth command and job. The Auth PR is pretty much ready to go, but I'm waiting to put it up for review until I finish writing automated tests for Web-E, and manual tests.

**Comment by neil-marcellini** [comment]
The implementation for updates when the policy changes is done, automated tests are passing, and the next step is testing on NewDot manually to make sure that generally works as expected.

Heads up I will be OOO on Monday.

**Comment by neil-marcellini** [comment]
I got PRs up for review today for the jobs and command to update reports when the policy updates. @mjasikowski is going to be taking this over while I'm OOO July 4-16, so I'm planning to sync up with him tomorrow and go over what needs to be done.

**Comment by neil-marcellini** [comment]
He is [that update](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1751491021709089?thread_ts=1749570249.246759&cid=C07NZ8B1VTQ) as promised.

**Comment by mjasikowski** [comment]
Status update: I've pushed a Web PR to re-enable authBetas and an App PR to display custom report title name errors. These are both in review now.
In the meantime, I've asked for volunteers again to cover for https://github.com/Expensify/App/issues/59689#top in https://expensify.slack.com/archives/C08CZDJFJ77/p1752063328495319
Will be tackling https://github.com/Expensify/Expensify/issues/520364 ASAP, possibly tomorrow

**Comment by mjasikowski** [comment]
The PR to re-enable authBetas was merged.
Waiting for reviews on https://github.com/Expensify/App/pull/64008, so we can push https://github.com/Expensify/Web-Expensify/pull/47167.
Starting work on https://github.com/Expensify/Expensify/issues/520364 today.
Meanwhile, Hubert resumed his work on https://github.com/Expensify/App/issues/59689

**Comment by mjasikowski** [comment]
Continuing work on https://github.com/Expensify/Expensify/issues/520364

**Comment by mjasikowski** [comment]
PR for https://github.com/Expensify/Expensify/issues/520364 was submitted for review today
Also fixed https://github.com/Expensify/Web-Expensify/pull/47167 so it doesn't allow strings not containing a formula

**Comment by neil-marcellini** [comment]
I'm back. Thanks for leading it while I was out @mjasikowski! I will take back over the lead from here and start re-assigning myself to some issues.

**Comment by neil-marcellini** [comment]
Yesterday, I got the pull request up to deprecate formulas. It ended up being fairly involved to get the exact behavior we are after, but I think it will be solid. I also learned that when classic shows errors for formulas, it still saves them. So the deprecation is really quite soft. We're basically just adding additional warnings on classic, but preventing these formulas being set from New Expensify.

Next up I will keep working on [Fix ComputeReportNames overriding manually set titles](https://github.com/Expensify/Expensify/issues/520364#top)

**Comment by neil-marcellini** [comment]
Not overdue, it's Monday.

**Comment by neil-marcellini** [comment]
See the [latest update here](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1753814497710659?thread_ts=1753808364.354769&cid=C07NZ8B1VTQ)

**Comment by neil-marcellini** [comment]
Today I updated my PR for [Fix ComputeReportNames overriding manually set titles](https://github.com/Expensify/Expensify/issues/520364#top), fixed one diff for [Monitor and resolve differences in Auth report name computation](https://github.com/Expensify/Expensify/issues/510023#top), and did an initial PR review for [Optimistic report title formula computation](https://github.com/Expensify/App/issues/59689). I also spent a lot of time determining the data formats we should support.

**Comment by neil-marcellini** [comment]
I'm reviewing the optimistic title PR today

**Comment by neil-marcellini** [comment]
Still moving along with that

**Comment by neil-marcellini** [comment]
Ok thanks for letting me know. I will grab that one from you since it's the most acute issue, and whenever you have time please jump in on this one, creating a sub issue for the type of problem you find in the logs. https://github.com/Expensify/Expensify/issues/510023

**Comment by neil-marcellini** [comment]
Check out yesterday's update [in Slack here](https://expensify.slack.com/archives/C07NMDKEFMH/p1756157520472099?thread_ts=1755619308.454669&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
Today's update also in that thread

**Comment by neil-marcellini** [comment]
[Todays update](https://expensify.slack.com/archives/C07NMDKEFMH/p1756432819983589?thread_ts=1755619308.454669&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1756523347558489?thread_ts=1755619308.454669&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1756764658815779?thread_ts=1755619308.454669&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1756933891447919?thread_ts=1756847765.319609&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1757023534136959?thread_ts=1756847765.319609&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1757453806316469?thread_ts=1757436716.938039&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1757541937605199?thread_ts=1757436716.938039&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1758051959439949?thread_ts=1758051892.644179&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1758151146543719?thread_ts=1758051892.644179&cid=C07NMDKEFMH), approaching launch!

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1758315845954919?thread_ts=1758051892.644179&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1758583587857379?thread_ts=1758051892.644179&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1758832968491929?thread_ts=1758644834.426199&cid=C07NMDKEFMH). We are starting to roll out the backend feature!

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1758919179980809?thread_ts=1758644834.426199&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1759268109800219?thread_ts=1758644834.426199&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1759444787811259?thread_ts=1758644834.426199&cid=C07NMDKEFMH). Only one blocker remains.

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1759521685701039?thread_ts=1758644834.426199&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
https://expensify.slack.com/archives/C07NMDKEFMH/p1759872113605619?thread_ts=1759830707.012519&cid=C07NMDKEFMH

**Comment by neil-marcellini** [comment]
https://expensify.slack.com/archives/C07NMDKEFMH/p1760493866051129?thread_ts=1760379074.787409&cid=C07NMDKEFMH

**Comment by neil-marcellini** [comment]
[The grind continues](https://expensify.slack.com/archives/C07NMDKEFMH/p1760666726251039?thread_ts=1760379074.787409&cid=C07NMDKEFMH) 👊

**Comment by neil-marcellini** [comment]
https://expensify.slack.com/archives/C07NMDKEFMH/p1761008715986229?thread_ts=1760987324.284159&cid=C07NMDKEFMH

**Comment by neil-marcellini** [comment]
Yesterday's update [was here](https://expensify.slack.com/archives/C07NMDKEFMH/p1761597371765989?thread_ts=1761597364.506969&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1762192906347309?thread_ts=1762192886.054239&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
https://expensify.slack.com/archives/C07NMDKEFMH/p1762393046149639?thread_ts=1762192886.054239&cid=C07NMDKEFMH

**Comment by neil-marcellini** [comment]
https://expensify.slack.com/archives/C07NMDKEFMH/p1762470555465649?thread_ts=1762192886.054239&cid=C07NMDKEFMH

**Comment by neil-marcellini** [comment]
We [launched the backend feature](https://github.com/Expensify/Auth/pull/17624#issuecomment-3504554762) on 11/7/25 at 11:40am PST!

Now we're just finishing off the offline optimistic computation. [Today's update](https://expensify.slack.com/archives/C01GTK53T8Q/p1762806302231949?thread_ts=1760387161.873509&cid=C01GTK53T8Q).

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1763747062387039?thread_ts=1763406260.585259&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1764895310425629?thread_ts=1764617063.870479&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1764971901116039?thread_ts=1764617063.870479&cid=C07NMDKEFMH), we will probably attempt another launch on Monday.

**Comment by neil-marcellini** [comment]
We launched the feature again [yesterday at 3:31pm PST](https://expensify.slack.com/archives/C07NMDKEFMH/p1765323081369909?thread_ts=1765227237.906639&cid=C07NMDKEFMH) 🚀

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1765416804385939?thread_ts=1765227237.906639&cid=C07NMDKEFMH). The polish and clean up phase starts 🧹

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1765577270419469?thread_ts=1765227237.906639&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
https://expensify.slack.com/archives/C07NMDKEFMH/p1767223927259639?thread_ts=1765820866.824369&cid=C07NMDKEFMH

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1767834908968559?thread_ts=1767663359.996789&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
Yesterday's update [was here](https://expensify.slack.com/archives/C07NMDKEFMH/p1770083110735709?thread_ts=1769475508.263849&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
## Daily Update - Feb 5, 2026

**Key milestones:**
- **App PR #81428** (Remove custom report names beta gating) merged yesterday - feature is now live for all users
- **Auth PR #19729** (Remove isOnAuthAutoReportTitleBeta) also merged yesterday

**Unblocked today:**
- **Web PR #50564** (Remove NE custom report name beta) - hold removed, ready for CI/review
- **Web PR #50558** (Remove isOnAuthAutoReportTitleBeta parameter) - hold removed, ready for CI/review

**Reviews completed:**
- PR #81504 (Restrict usage of computeReportName) - reviewed with minor suggestions
- PR #81513 (Simplify computation of report name) - changes requested, waiting on author
- PR #81322 (Fix negative amount in expense title) - commented that the formula path handles this after our changes
- Issue #80045 (Description Hint field) - approved MobileMage's proposal

**Still open:**
- App PR #80254 (Allow editing report titles if title field is missing)
- Web PR #50476 (Add script to generate expenses with categories)

**Comment by neil-marcellini** [comment]
Polishing up the last details - [today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1770332885612529?thread_ts=1770139028.323969&cid=C07NMDKEFMH)

**Comment by neil-marcellini** [comment]
[Today's update](https://expensify.slack.com/archives/C07NMDKEFMH/p1770844701669689?thread_ts=1770682030.679009&cid=C07NMDKEFMH)
